Description: Remove the Built-At, Built-By and Built-On entries in the manifest of the jar files to improve the build reproducibility
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: no
--- a/build.xml
+++ b/build.xml
@@ -550,9 +550,6 @@
             <fileset dir="${src_generated.dir}" excludes="**/.generated"/>
             <manifest>
                 <attribute name="Main-Class" value="org.apache.zookeeper.server.quorum.QuorumPeer" />
-                <attribute name="Built-By" value="${user.name}"/>
-                <attribute name="Built-At" value="${build.time}"/>
-                <attribute name="Built-On" value="${host.name}" />
                 <attribute name="Implementation-Title" value="org.apache.zookeeper"/>
                 <attribute name="Implementation-Version" value="${revision}"/> 
                 <attribute name="Implementation-Vendor" value="The Apache Software Foundation"/>
@@ -591,9 +588,6 @@
             <fileset dir="${build.classes}" excludes="**/.generated"/>
             <manifest>
                 <attribute name="Main-Class" value="org.apache.zookeeper.server.quorum.QuorumPeer" />
-                <attribute name="Built-By" value="${user.name}"/>
-                <attribute name="Built-At" value="${build.time}"/>
-                <attribute name="Built-On" value="${host.name}" />
                 <attribute name="Implementation-Title" value="org.apache.zookeeper"/>
                 <attribute name="Implementation-Version" value="${revision}"/> 
                 <attribute name="Implementation-Vendor" value="The Apache Software Foundation"/>
@@ -623,9 +617,6 @@
             <fileset dir="${java.src.dir}"/>
             <fileset dir="${src_generated.dir}" excludes="**/.generated"/>
             <manifest>
-                <attribute name="Built-By" value="${user.name}"/>
-                <attribute name="Built-At" value="${build.time}"/>
-                <attribute name="Built-On" value="${host.name}" />
                 <attribute name="Implementation-Title" value="org.apache.zookeeper"/>
                 <attribute name="Implementation-Version" value="${revision}"/> 
                 <attribute name="Implementation-Vendor" value="The Apache Software Foundation"/>
@@ -641,9 +632,6 @@
             <fileset file="LICENSE.txt" />
             <fileset dir="${build.javadoc}"/>
             <manifest>
-                <attribute name="Built-By" value="${user.name}"/>
-                <attribute name="Built-At" value="${build.time}"/>
-                <attribute name="Built-On" value="${host.name}" />
                 <attribute name="Implementation-Title" value="org.apache.zookeeper"/>
                 <attribute name="Implementation-Version" value="${revision}"/> 
                 <attribute name="Implementation-Vendor" value="The Apache Software Foundation"/>
@@ -659,9 +647,6 @@
             <fileset file="LICENSE.txt" />
             <fileset dir="${test.java.classes}"/>
             <manifest>
-                <attribute name="Built-By" value="${user.name}"/>
-                <attribute name="Built-At" value="${build.time}"/>
-                <attribute name="Built-On" value="${host.name}" />
                 <attribute name="Implementation-Title" value="org.apache.zookeeper"/>
                 <attribute name="Implementation-Version" value="${revision}"/> 
                 <attribute name="Implementation-Vendor" value="The Apache Software Foundation"/>
